Automattic/jetpack
# Sharing Block: automatically add block to templates in block themes

> [!IMPORTANT]
> This is an experiment to play with the Block Hooks API. It can only work with recent changes made to WP Core, and scheduled to ship in WordPress 6.5 next month.

## Proposed changes:

Under a few specific conditions, we want to automatically add a sharing buttons block (and 3 inner blocks, Facebook, X, and Mastodon) to the bottom of the post content.
Site owners can then go to Appearance > Editor if they want to remove or customize that block.

For now, we're trying to automatically add the block right after the post content block, in post and page templates (that includes templates like `page-wide`, `page-no-sidebar`, etc).

Here are the conditions:

- The site must be using a block theme.
- An option (`jetpack_sharing_buttons_auto_add`) must be set. Later on we can add UI to set that option under Jetpack > Settings > Social.
- The sharing module must be deactivated. We wouldn't want to automatically add sharing buttons to the post content twice.
- On WordPress.com Simple, the module is always active. There, we check if sharing posts are enabled on single posts and / or pages, since that matches the template where we'd like to automatically add the block.

**To do**

- [x] This doesn't quite work just yet, I haven't been able to figure out how to insert inner blocks. The site editor currently shows a validation error when viewing the single post template:
- [x] We should figure out how this should play with the sharedaddy module. Should we look at sharedaddy settings like we do right now and base our response on that? Or should we not look at it at all? In this case, when should auto-addition be triggered on WordPress.com Simple, where the sharing module is always active?

### Resources:

- https://make.wordpress.org/core/2023/10/15/introducing-block-hooks-for-dynamic-blocks/
- https://github.com/ockham/like-button
- https://core.trac.wordpress.org/ticket/59572
- https://core.trac.wordpress.org/ticket/60126

### Other information:


## Jetpack product discussion

- p1707216098587689-slack-C065QF2R1B6

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Start by installing the development version of WordPress. Alternatively, [launch a quick test site via this link](https://jurassic.ninja/create?jetpack-beta&branches.jetpack=add/hook-sharing-block&wp-debug-log&wordpress-beta-tester&gutenberg).
* Go to Appearance > Themes and install a block-based theme.
* Go to Jetpack > Settings > Sharing and ensure that the Sharing buttons module is **off**.
* Via WP CLI, set the `jetpack_sharing_buttons_auto_add` option: `wp option add jetpack_sharing_buttons_auto_add true`
* Visit a single post on the frontend
    * Sharing buttons (Facebook, X, Mastodon) should be added below the post content.
